sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -2241,6 +2246,14 @@ static String getExtractDestinationUri(String extractDestinationDir) {
       return toBuilder().setIgnoreUnknownValues(true).build();
     }
 
+    /**
+     * Performs streaming insert without insert id. Insert id is used to offer best effort insert
+     * deduplication. Default is false, which always inserts with insert id.
+     */
+    public Write<T> ignoreInsertIds() {
```

Comment:
What kind of guarantees can we provide regarding duplicate data inserts when this option is set ? My understanding is that this will prevent BQ from data deduplication at all hence users may start observing duplicate data in the output table when there are workitem failures and retries. Even though insert ID based data deduplication is best-effort, I think, in practice, that is enough for most users. 

I think this option at least should come with a warning that says that setting this will completely disable data deduplication a when inserting records to BigQuery and that users should utilize custom post-insertion data duplication mechanisms if needed. 

cc: @reuvenlax

Reply:
Maybe rename withIgnoreInsertId for consistency?

Reply:
Ignore seems to be the verb/action itself, I just made it consistent with ignoreUnknownFields.